unumpy/tests/test_numpy.py

```diff
@@ -205,12 +208,29 @@ def test_multiple_output(backend, method, args, kwargs):
 )
 def test_array_creation(backend, method, args, kwargs):
     backend, types = backend
-    try:
```

Comment:
Why only the NumPy backend?

Reply:
Tests were failing for `xnd_backend` currently.
```
______________ test_array_creation[backend4-zeros-args0-kwargs0] _______________

backend = <module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
method = <uarray multimethod 'zeros'>, args = ((1, 2, 3),)
kwargs = {'dtype': <class 'numpy.int8'>}

    @pytest.mark.parametrize(
        "method, args, kwargs",
        [
            (np.zeros, ((1, 2, 3),), {}),
            (np.ones, ((1, 2, 3),), {}),
            (np.full, ((1, 2, 3), 1.3), {}),
        ],
    )
    def test_array_creation(backend, method, args, kwargs):
        backend, types = backend
        # if backend == NumpyBackend:
        for dtype in dtypes:
            try:
                with ua.set_backend(backend, coerce=True):
                    kwargs["dtype"] = dtype
                    ret = method(*args, **kwargs)
            except ua.BackendNotImplementedError:
                if (
                    backend in FULLY_TESTED_BACKENDS
                    and (backend, method) not in EXCEPTIONS
                ):
                    raise
                pytest.xfail(reason="The backend has no implementation for this ufunc.")
    
            assert isinstance(ret, types)
            print(backend)
>           assert ret.dtype == dtype
E           assert ndt("int8") == <class 'numpy.int8'>
E            +  where ndt("int8") = <[ValueError('The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()') raised in repr()] array object at 0x7f633d4f0350>.dtype

unumpy/tests/test_numpy.py:227: AssertionError
----------------------------- Captured stdout call -----------------------------
<module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
_______________ test_array_creation[backend4-ones-args1-kwargs1] _______________

backend = <module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
method = <uarray multimethod 'ones'>, args = ((1, 2, 3),)
kwargs = {'dtype': <class 'numpy.int8'>}

    @pytest.mark.parametrize(
        "method, args, kwargs",
        [
            (np.zeros, ((1, 2, 3),), {}),
            (np.ones, ((1, 2, 3),), {}),
            (np.full, ((1, 2, 3), 1.3), {}),
        ],
    )
    def test_array_creation(backend, method, args, kwargs):
        backend, types = backend
        # if backend == NumpyBackend:
        for dtype in dtypes:
            try:
                with ua.set_backend(backend, coerce=True):
                    kwargs["dtype"] = dtype
                    ret = method(*args, **kwargs)
            except ua.BackendNotImplementedError:
                if (
                    backend in FULLY_TESTED_BACKENDS
                    and (backend, method) not in EXCEPTIONS
                ):
                    raise
                pytest.xfail(reason="The backend has no implementation for this ufunc.")
    
            assert isinstance(ret, types)
            print(backend)
>           assert ret.dtype == dtype
E           assert ndt("int8") == <class 'numpy.int8'>
E            +  where ndt("int8") = <[ValueError('The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()') raised in repr()] array object at 0x7f633d4f0a50>.dtype

unumpy/tests/test_numpy.py:227: AssertionError
----------------------------- Captured stdout call -----------------------------
<module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
_______________ test_array_creation[backend4-full-args2-kwargs2] _______________

backend = <module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
method = <uarray multimethod 'full'>, args = ((1, 2, 3), 1.3)
kwargs = {'dtype': <class 'numpy.int8'>}

    @pytest.mark.parametrize(
        "method, args, kwargs",
        [
            (np.zeros, ((1, 2, 3),), {}),
            (np.ones, ((1, 2, 3),), {}),
            (np.full, ((1, 2, 3), 1.3), {}),
        ],
    )
    def test_array_creation(backend, method, args, kwargs):
        backend, types = backend
        # if backend == NumpyBackend:
        for dtype in dtypes:
            try:
                with ua.set_backend(backend, coerce=True):
                    kwargs["dtype"] = dtype
                    ret = method(*args, **kwargs)
            except ua.BackendNotImplementedError:
                if (
                    backend in FULLY_TESTED_BACKENDS
                    and (backend, method) not in EXCEPTIONS
                ):
                    raise
                pytest.xfail(reason="The backend has no implementation for this ufunc.")
    
            assert isinstance(ret, types)
            print(backend)
>           assert ret.dtype == dtype
E           assert ndt("int8") == <class 'numpy.int8'>
E            +  where ndt("int8") = <[ValueError('The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()') raised in repr()] array object at 0x7f633d4f0f50>.dtype

unumpy/tests/test_numpy.py:227: AssertionError
----------------------------- Captured stdout call -----------------------------
<module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
```

Reply:
Looking into it. Will modify the PR accordingly.

Reply:
@hameerabbasi , Please review again :)